File: cpp/1993-sum-of-all-subset-xor-totals.cpp
class Solution {
public:
    int subsetXORSum(vector<int>& nums) {
        return helper(nums, 0, 0);
    }
    int helper(vector<int>& nums, int index, int currentXOR) {
        if (index == nums.size()) {
            return currentXOR;
        }
        int with = helper(nums, index + 1, currentXOR ^ nums[index]);
        int without = helper(nums, index + 1, currentXOR);
        return with + without;
    }
};

File: go/1993-sum-of-all-subset-xor-totals.go
func subsetXORSum(nums []int) int {
    var dfs func(int, int) int
    dfs = func(index, currentXOR int) int {
        if index == len(nums) {
            return currentXOR
        }
        with := dfs(index+1, currentXOR^nums[index])
        without := dfs(index+1, currentXOR)
        return with + without
    }
    return dfs(0, 0)
}
